addons/html_editor/static/src/editor/core/delete_plugin.js

```diff
@@ -0,0 +1,726 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { Plugin } from "../plugin";
+import { closestBlock, isBlock } from "../utils/blocks";
+import { fillEmpty, moveNodes } from "../utils/dom";
+import {
+    isEditorTab,
+    isEmptyBlock,
+    isIconElement,
+    isInPre,
+    isMediaElement,
+    isNotEditableNode,
+    isSelfClosingElement,
+    isUnbreakable,
+    isUnremovable,
+    isVisible,
+    isVisibleTextNode,
+    isWhitespace,
+    isZWS,
+    paragraphRelatedElements,
+} from "../utils/dom_info";
+import { getState, prepareUpdate } from "../utils/dom_state";
+import { closestElement, createDOMPathGenerator, findNode } from "../utils/dom_traversal";
+import {
+    DIRECTIONS,
+    boundariesOut,
+    childNodeIndex,
+    endPos,
+    leftPos,
+    nodeSize,
+    rightPos,
+} from "../utils/position";
+import { getDeepRange, setSelection } from "../utils/selection";
+import { CTGROUPS, CTYPES } from "../utils/content_types";
+import { splitTextNode } from "../utils/dom_split";
+
+const shortcuts = {
+    DELETE_BACKWARD: (e) => e.key === "Backspace" && !e.ctrlKey && !e.metaKey,
+};
+
+export class DeletePlugin extends Plugin {
+    static dependencies = ["dom"];
+    static name = "delete";
+
+    setup() {
+        this.addDomListener(this.editable, "keydown", this.handleShortcut.bind(this));
+        this.addDomListener(this.editable, "beforeinput", this.onBeforeInput.bind(this));
+    }
+    handleCommand(command, payload) {
+        switch (command) {
+            case "DELETE_BACKWARD":
+                this.deleteBackward();
+                break;
+            case "DELETE_FORWARD":
+                this.deleteForward();
+                break;
+            case "DELETE_ELEMENT_BACKWARD":
+                this.deleteElementBackward(payload);
+                break;
+            case "DELETE_ELEMENT_FORWARD":
+                this.deleteElementForward(payload);
+                break;
+            case "DELETE_RANGE":
+                this.deleteRange();
+                break;
+        }
+    }
+
+    // --------------------------------------------------------------------------
+    // commands
+    // --------------------------------------------------------------------------
+
+    deleteBackward() {
+        let selection = this.shared.getEditableSelection();
+        if (!selection) {
+            return;
+        }
+        if (handleZWS(this.editable, selection)) {
+            this.dispatch("DELETE_RANGE");
+            return;
+        }
+        selection = this.shared.getEditableSelection();
```

Comment:
it looks like this line can be removed, since handleZWS returned false and did not modify the editable selection (??)

Reply:
No because 

```
getDeepRange(editable, {
        selection,
        splitText: true,
        select: true,
        correctTripleClick: true,
    });
```
will effectively change the selection with `select: true`